test.sh

```diff
@@ -146,7 +146,18 @@ GOBIN=${GOBIN:-$HOME/gopath/bin}
 #
 if [[ "$RUN" =~ "vet" ]] ; then
   start_context "test/vet"
-  run_and_comment go vet ./...
+  check_go_vet() {
```

Comment:
I'm still confused about what you're thinking here.  `run` looks for whether the function outputs 0 or 1 and sets the status accordingly.  `run_and_comment` calls `run` to get the output and set the status value, then comments if required.  If `run_and_comment` returns 1, there's no point -- it never gets read!  I think what you mean to say is that `run_and_comment` should set the status, but that's duplicative of what's going on in `run`.

Instead, I've added an inner wrapper function that takes non-empty output and turns that into a non-zero status.  Then `run` and `run_and_comment` should do what's expected.